usermods/battery_status_basic/usermod_v2_battery_status_basic.h

```diff
@@ -0,0 +1,260 @@
+  private:
+    // battery pin can be defined in my_config.h
+    int8_t batteryPin = BATTERY_MEASUREMENT_PIN;
+    // how often to read the battery voltage
+    unsigned long readingInterval = USERMOD_BATTERY_MEASUREMENT_INTERVAL;
+    unsigned long lastTime = 0;
+    // battery min. voltage
+    float minBatteryVoltage = USERMOD_BATTERY_MIN_VOLTAGE;
+    // battery max. voltage
+    float maxBatteryVoltage = USERMOD_BATTERY_MAX_VOLTAGE;
+    float rawValue = 0.0;
+    // calculated voltage            
+    float voltage = 0.0;
+    // mapped battery level based on voltage
+    long batteryLevel = 0;
+    
+    /*
+    // set your config variables to their boot default value (this can also be done in readFromConfig() or a constructor if you prefer)
+    bool testBool = false;
+    unsigned long testULong = 42424242;
+    float testFloat = 42.42;
+    String testString = "Forty-Two";
+
+    // These config variables have defaults set inside readFromConfig()
+    int testInt;
+    long testLong;
+    int8_t testPins[2];
+    */
+
+  public:
+    //Functions called by WLED
+
+    // https://forum.arduino.cc/t/floating-point-using-map-function/348113/2
+    double mapf(double x, double in_min, double in_max, double out_min, double out_max) {
+      return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min;
+    }
+
+    /*
+     * setup() is called once at boot. WiFi is not yet connected at this point.
+     * You can use it to initialize variables, sensors or similar.
+     */
+    void setup() {
+      pinMode(batteryPin, INPUT);
+    }
+
+
+    /*
+     * connected() is called every time the WiFi is (re)connected
+     * Use it to initialize network interfaces
+     */
+    void connected() {
+      //Serial.println("Connected to WiFi!");
+    }
+
+
+    /*
+     * loop() is called continuously. Here you can check for events, read sensors, etc.
+     * 
+     */
+    void loop() {
+      unsigned long now = millis();
+
+      // check the battery level every USERMOD_BATTERY_MEASUREMENT_INTERVAL (ms)
+      if (now - lastTime >= readingInterval) {
+
+        // read battery raw input
+        rawValue = analogRead(batteryPin);
+
+        // calculate the voltage     
+        voltage = (rawValue / 1024.0) * maxBatteryVoltage ;
+
+        // translate battery voltage to a percentage
+        /*
+          the standard "map" function doesn't work
+          https://www.arduino.cc/reference/en/language/functions/math/map/ <-- notes and warnings at the bottom
+        */
+        batteryLevel = mapf(voltage, minBatteryVoltage, maxBatteryVoltage, 0, 100);
+
+        lastTime = now;
+      }
+    }
+
+
+    /*
+     * addToJsonInfo() can be used to add custom entries to the /json/info part of the JSON API.
+     * Creating an "u" object allows you to add custom key/value pairs to the Info section of the WLED web UI.
+     * Below it is shown how this could be used for e.g. a light sensor
+     */
+    void addToJsonInfo(JsonObject& root)
```

Comment:
You may need to rewrite this to make it compatible with _standard_ Info page.

Reply:
done.
will now take care of the other things